### PR TITLE
[Bug fix] : FIlePicker - if Parse content is disabled, the "File type" dropdown should be hidden 

### LINKED ISSUE #3621: [Bug] : FilePicker widget - parse content
### Version Information
v1.20.2
### Environment
Development

### What is the expected behavior?
- If Parse content is disabled, the "File type" dropdown should be hidden
- parse data when Parse content is set to true or toggled on

### What is the current behavior?
- Parse content is disabled, and the File type dropdown options are not hidden- 
- Even if the Parse content toggled is disabled the file gets parsed.

### How to reproduce the issue?

1. Drag the FilePicker widget on the canvas
2. Open the widget inspector by clicking on the label of the dragged file picker widget on the canvas
3. Properties tab will be shown on the widget inspector (left side-bar), scroll down and look for the `Options` block
4. Inside it if parse content is disabled, the file type option should be hidden, which is not the case, it's always showing  

### Screenshots or Screencast
Screenshot 2022-07-15 at 4_Fotor

### PR DESCRIPTION
**Pull request by manishkushare**
Resolve #3621
Fixed: If Parse content is disabled, the "File type" dropdown should be hidden

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2kbkz8p If Parse content is disabled,  "File type" dropdown should be hidden and don't parse the data](https://app.clickup.com/t/2kbkz8p)

**Comment by Nandinisaha13** [comment]
<html>
<body>
<google-sheets-html-origin>Test Case

Test Cases | Steps | Result
-- | -- | --
Should not be able to see file types while parse content is disabled | Create an application. | Application gets created.
  | Drag and drop file picker. | File picker is displayed on the canvas.
  | Go to parse content in properties. | Parse content is visible.
  | Toggle to disable parse content. | The file type is hidden.


</body>
</html>

File: frontend/src/Editor/Inspector/Components/FilePicker.jsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ export const FilePicker = ({ componentMeta, darkMode, ...restProps }) => {
     const conditionalOptions = [{ name: 'parseFileType', condition: parseContent }];
```
Comment (arpitnath) [inline frontend/src/Editor/Inspector/Components/FilePicker.jsx:30]: Can we check the condition here: 
`const conditionalOptions = [{ name: 'parseFileType', condition: parseContent === '{{true}}' }];`

File: frontend/src/Editor/Inspector/Components/FilePicker.jsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export const FilePicker = ({ componentMeta, darkMode, ...restProps }) => {
...
 
     conditionalOptions.map(({ name, condition }) => {
-      if (condition) renderOptions.push(renderCustomElement(name));
+      if (condition === '{{true}}') renderOptions.push(renderCustomElement(name));
```
Comment (arpitnath) [inline frontend/src/Editor/Inspector/Components/FilePicker.jsx:33]: And remove this?

File: frontend/src/Editor/Inspector/Components/FilePicker.jsx (hunk 1 above)
Comment (arpitnath) [inline frontend/src/Editor/Inspector/Components/FilePicker.jsx:30]: Can we check the condition here:
const conditionalOptions = [{ name: 'parseFileType', condition: parseContent === '{{true}}' }];

**Review by arpitnath — CHANGES_REQUESTED** [review]

**Comment by shubh22** [comment]
Task linked: [CU-2mtjqb9 Filepicker: [Bugfix]](https://app.clickup.com/t/2mtjqb9)

**Review by arpitnath — DISMISSED** [review]
LGTM!

File: frontend/src/Editor/Inspector/Components/FilePicker.jsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ export const FilePicker = ({ componentMeta, darkMode, ...restProps }) => {
 
     options.map((option) => renderOptions.push(renderCustomElement(option)));
 
-    const conditionalOptions = [{ name: 'parseFileType', condition: parseContent }];
+    const conditionalOptions = [{ name: 'parseFileType', condition: parseContent === '{{true}}' }];
```
Comment (sherfin94) [inline frontend/src/Editor/Inspector/Components/FilePicker.jsx:30]: If `parseContent` is programmatically configured (eg: `variables.someVariable > 33.3`), this condition will be erroneous won't it?

Resolve it using resolveReferences and then check if it is `true`.

**Review by sherfin94 — CHANGES_REQUESTED** [review]
One issue reported on code, please check

**Comment by manishkushare** [comment]
@sherfin94 , Made the suggested changes.

**Review by kavinvenkatachalam — COMMENTED** [review]
Suggestion - Unless the value of `component.definition.properties.parseContent?.value` is undefined, `resolveWidgetFieldValue` function will be called. Can we re-write it in a single line?
cc: @sherfin94 @arpitnath

File: frontend/src/Editor/Components/FilePicker.jsx [hunk 4]
```diff
@@ -24,9 +24,9 @@ export const FilePicker = ({
   const fileType = component.definition.properties.fileType?.value ?? 'image/*';
   const maxSize = component.definition.properties.maxSize?.value ?? 1048576;
   const minSize = component.definition.properties.minSize?.value ?? 0;
-  const parseContent = component.definition.properties.parseContent?.value ?? false;
+  let parseContent = component.definition.properties.parseContent?.value ?? false;
+  parseContent = typeof parseContent !== 'boolean' ? resolveWidgetFieldValue(parseContent, currentState) : false;
```
Comment (kavinvenkatachalam) [inline frontend/src/Editor/Components/FilePicker.jsx:28]:
```suggestion
  const parseContent = resolveWidgetFieldValue(component.definition.properties.parseContent?.value ?? false, currentState);
```

**Comment by manishkushare** [comment]
@kavinvenkatachalam , made the requested changes

**Review by kavinvenkatachalam — APPROVED** [review]
LGTM

**Review by sherfin94 — APPROVED** [review]

**Comment by shubh22** [comment]
Task linked: [CU-2k0aqfh File Picker Updates: Improvements and Bugfixes](https://app.clickup.com/t/2k0aqfh)

**Pull request merged**